src/Chapter1.hs

```diff
@@ -636,8 +651,12 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Integral a => a -> a
+firstDigit n = let 
+                   rest = n `div` 10
+               in
+                 if rest < 10 then rest else firstDigit rest
```

Comment:
I see, that for a negative number like -19, this implementation will return the negative number itself instead of the last digit because the first check n < 10 will always succeed for negative numbers.